tools/ncf_constraints.py

```diff
@@ -0,0 +1,111 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+if sys.version_info < (3,):
+  string_type = unicode
+else:
+  string_type = str
+
+### Constraint checking function
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
```

Comment:
re.S and re.M should be considered
It means that a parameter should be considered as a whole and we are matching it as is, not with separate lines. A new line can still be matched with \n, which is expected. And we can match things that couldn't be matched without those flags. 

Example:
- the regex r"^A.*\nB.*$" matches a line starting with A followed by a line starting with B. This is not possible without re.M
- with re.S r"A.*B.*" could match 2 lines with A in the first one and B in the second one. This may be counter intuitive. 

So I'd say, use re.MULTILINE only.

Anyway, whatever the final behavior, the multiline case must be carefully documented.